src/test/java/lotto/domain/WinningLottoTest.java

```diff
@@ -0,0 +1,31 @@
+package lotto.domain;
+
+import lotto.utils.Result;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WinningLottoTest {
+    @Test
+    void 로또번호일치() {
+        WinningLotto winningLotto = new WinningLotto(
+                new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6)),
+                new Digit(10));
+
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(5, 6, 7, 8, 9, 10)))).isEqualTo(Result.UNDERTHREE);
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(4, 5, 6, 7, 8, 9)))).isEqualTo(Result.THREE);
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(3, 4, 5, 6, 7, 8)))).isEqualTo(Result.FOUR);
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(2, 3, 4, 5, 6, 9)))).isEqualTo(Result.FIVE);
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(2, 3, 4, 5, 6, 10)))).isEqualTo(Result.BONUSFIVE);
+        assertThat(winningLotto.compare(new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6)))).isEqualTo(Result.SIX);
+    }
+
+    @Test
+    void create() {
+        WinningLotto winningLotto = new WinningLotto(new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6)));
+        assertThat(winningLotto)
+                .isEqualTo(new WinningLotto(new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6))));
+    }
```

Comment:
질문사항 : 원활한 테스트를 위해 추가한 생성자(혹은 메서드)는 그대로 두는 게 일반적인가요?
답변 : 그렇지 않습니다. 보통 테스트를 위한 생성자를 추가히지 않고, 로직에서 사용하는 생성자를 이용하여 테스트하는게 좋아요.
이 테스트는 아래와 같이 기존 생성자를 쓰도록 바꿔주면  충분히 테스트가 가능해요. 
테스트에서만 사용하는 생성자가 존재하면 다른 사람들이 오해할 소지가 있고 잘못 생성자를 택해서 사용할 경우 사이드이펙트가 발생할 수 있으니 지양해주세요. 
```suggestion
WinningLotto winningLotto = new WinningLotto(new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6)), new Digit(37));
        assertThat(winningLotto)
                .isEqualTo(new WinningLotto(new Lotto(Arrays.asList(1, 2, 3, 4, 5, 6)), new Digit(37)));